### ISSUE TITLE
salt-cloud --out-file option not work

### ISSUE THREAD

**Issue by ThomasZhou** [saltstack/salt#28281]
when call salt-cloud, set --out-file option, but found not work, nothing for the file.

salt-cloud --version
salt-cloud 2015.8.0 (Beryllium)

**Comment by jfindlay** [comment]
@ThomasZhou, thanks for the report.  What is the path to the file you were trying to output to and does the user you are using salt cloud with have write permission to that location?  Are there any debug log messages related to not finding or writing to the file?

**Comment by ThomasZhou** [comment]
We have rights to access the specified path.

1. salt command is ok
salt '*' network.netstat --out=json --out-file=/home/test/test.log

/home/test# ll
total 12
drwxr-xr-x 2 root root 4096 Nov  4 20:04 ./
drwxr-xr-x 9 root root 4096 Nov  4 20:03 ../
-rw-r--r-- 1 root root 2509 Nov  4 20:04 test.log

2. salt-cloud is not work for out-file option,not create the file,still output to console
salt-cloud -Q --out=json --out-file=/home/test/test2.log
/home/test# ll
total 12
drwxr-xr-x 2 root root 4096 Nov  4 20:04 ./
drwxr-xr-x 9 root root 4096 Nov  4 20:03 ../
-rw-r--r-- 1 root root 2509 Nov  4 20:04 test.log

**Comment by jfindlay** [comment]
@ThomasZhou, thanks for the extra information.

**Comment by bogdanr** [comment]
I confirm this problem with 2015.8.1-121-gc1ac330

**Comment by rallytime** [comment]
I have also been able to reproduce this bug.

**Comment by pass-by-value** [comment]
#29987 should fix this

**Comment by rallytime** [comment]
I can confirm that this is fixed at the HEAD of the 2015.8 branch:
```
root@rallytime:~# salt --versions
Salt Version:
           Salt: 2015.8.3-167-g5632ccb

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.10
          PyZMQ: 14.5.0
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.9.0
        timelib: Not Installed

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 4.1.5-x86_64-linode61
         system: Ubuntu 14.04 trusty

root@rallytime:~# salt-cloud --list-images linode --out=json
[INFO    ] salt-cloud starting
{
    "linode-config": {
        "linode": {
            "Fremont, CA, USA": {
                "DATACENTERID": 3,
                "LOCATION": "Fremont, CA, USA",
                "ABBR": "fremont"
            },
            "Dallas, TX, USA": {
                "DATACENTERID": 2,
                "LOCATION": "Dallas, TX, USA",
                "ABBR": "dallas"
            },
            "Singapore, SG": {
                "DATACENTERID": 9,
                "LOCATION": "Singapore, SG",
                "ABBR": "singapore"
            },
            "Tokyo, JP": {
                "DATACENTERID": 8,
                "LOCATION": "Tokyo, JP",
                "ABBR": "tokyo"
            },
            "London, England, UK": {
                "DATACENTERID": 7,
                "LOCATION": "London, England, UK",
                "ABBR": "london"
            },
            "Atlanta, GA, USA": {
                "DATACENTERID": 4,
                "LOCATION": "Atlanta, GA, USA",
                "ABBR": "atlanta"
            },
            "Newark, NJ, USA": {
                "DATACENTERID": 6,
                "LOCATION": "Newark, NJ, USA",
                "ABBR": "newark"
            },
            "Frankfurt, DE": {
                "DATACENTERID": 10,
                "LOCATION": "Frankfurt, DE",
                "ABBR": "frankfurt"
            }
        }
    }
}
```
This fix will be available in the upcoming 2015.8.4 release.
